docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
Wow, what a bizarre API.  I don't see what this extra class brings.  The obvious API would be `tlsProtocols={TLSv1_0, TLSv1_1, TLSv1_2}` with a helper API for generating a whole range - `tlsProtocols=tlsVersionRange(TLSv1_0, TLSv1_2)` (just returning a set).

If the docs explained the `Highest` feature, this would look a little bit less weird.  I'd say, at a minimum, the docs must explain that feature and encourage its use.  In fact, they should start with an example that shows what most people should do most of the time and then dive into obscure cases like the two examples given here.

But I'd still probably opt for a simpler set-based API.  Twisted can expose a `GOOD_TLS_VERSIONS` constant that does the same job as `Highest` does now - without the weird `TLSVersionRange` thing.  Failing that, `TLSVersionRange` should implement some formal interface instead of being a simple concrete type with some hard-coded logic.  This at least brings some functionality with the extra complexity (allowing implementation of dynamic protocol selection logic) which might justify it.